src/properties.jl

```diff
@@ -214,8 +171,80 @@ function Base.setproperty!(t::Task, channel::String, property::String, value)
     kind = channel_types[ findall(channel_type(t, channel)[1] .==
             map((x)->getfield(NIDAQ,Symbol(x)), channel_types))[1]][end-1:end]
 
-    @eval ret = $(Symbol("DAQmxSet"*kind*property))($t.th, Ref(codeunits($channel),1), $value)
+    @eval ret = $(Symbol("DAQmxSet"*kind*property))($t.th, str2code($channel), $value)
     catch_error(ret, "DAQmxSet$kind$property: ")
     nothing
 end
 
+
+function Base.getproperty(t::Task,channel::String,property::String)
+    kind = channel_types[ findall(channel_type(t, channel)[1] .==
+            map((x)->getfield(NIDAQ,Symbol(x)), channel_types))[1]][end-1:end]
+
+    cfunction = getfield(NIDAQ, Symbol("DAQmxGet"*kind*property))
+    data = getproperty((t.th, str2code(channel)),cfunction)
+    return data
+end
+
+function Base.getproperty(args,cfunction; warning=false)
+    ccall_args = code_lowered(cfunction)[1].code[end-1].args[3]
+    data = 0
+    try
+        basetype = eltype(ccall_args[1+length(args)])
+        if length(ccall_args)==1+length(args)
+            data = Ref{basetype}(0)
+            ret = cfunction(args..., data)
+            data = data[]
+        else
+            sz = cfunction(args..., convert(Ptr{basetype},C_NULL), convert(UInt32,0))
+            if sz<0
+              ret=sz
+              throw()
+            end
+            data = zeros(basetype,sz)
+            ret = cfunction(args..., Ref(data,1), convert(UInt32,sz))
+        end
+        if ret!=0
+            throw()
+        elseif basetype == Bool32
+            data = reinterpret(UInt32, data) != 0
+        elseif basetype == Int32
+            try
+                data = map((x)->signed_constants[x], data)
+            catch
+            end
+        elseif basetype == UInt32
+            try
+                data = map((x)->unsigned_constants[x], data)
+            catch
+            end
+        elseif basetype == UInt8
+            data = split(safechop(ascii(String(data))),", ")
+        elseif basetype == Int8
+            data = split(safechop(ascii(String(UInt8.(data)))),", ")
+
+        end
+    catch
+        if warning
+            if ret!=0
+                catch_error(ret, string(cfunction)*": ", err_fcn=x->@warn(x))
+            else
+                @warn("can't handle function signature for $cfunction: $ccall_args")
+            end
+        end
+        
+    end
+
+    return data
+
+end
+
+function str2code(s::String)
+    if NIDAQ.dType == Cchar
+        code = Ref(Cchar.(codeunits(s)),1)
+    elseif NIDAQ.dType == UInt8
+        code = Ref(codeunits(s),1)
+    end
+    return code
+end
```

Comment:
all references to the variable `code` could be removed here and this function would work identically:

```
function str2code(s::String)
    if NIDAQ.dType == Cchar
        Ref(Cchar.(codeunits(s)),1)
    elseif NIDAQ.dType == UInt8
        Ref(codeunits(s),1)
    else
        error("unsupported NIDAQ.dtype")
    end
end```

Reply:
note that you should also handle the case for which `NIDAQ.dType` is neither `Cchar` nor `UInt8`.

Reply:
or actually, since `NIDAQ.dType` is a global constant (right?), sth like this:

```
str2code(s::String) = str2code(Val(NIDAQ.dType), s)
str2code(::Val{Cchar}, s::String) = Ref(Cchar.(codeunits(s)),1)
str2code(::Val{UInt8}, s::String) = Ref(codeunits(s),1)
```

and then you can leave all your calls to `str2code` as is.